### ISSUE TITLE
Slow training speed

### ISSUE THREAD

**Issue by lsw5835** [OlaWod/FreeVC#61]
Hi, Is there a reason why learning speed decreases when the amount of learning data increases?

I understand the part where function "train_loader.batch_sampler.set_epoch(epoch)" is affected by datasetsize, but it is difficult to understand that it slows down in the part where the batch is taken from the train loader (in "for batch_idx, items in enumerate(train_loader):"). Is there any other reason to slow down?

Thanks very much.

**Comment by lsw5835** [comment]
I found that when trainloder get batches, they are going to "iter function" in "data_utils_24.py". Then the code deterministically shuffle and generate batches. Is there any reason that the code shuffle all batch and make batches list when they load one batch?

**Comment by OlaWod** [comment]
do you mean the convergence speed? more data means more features to learn, just like training a multi-speaker tts would take more time than single-speaker tts. also the data complexity can affect speed too, for example highly emotional data.

**Comment by lsw5835** [comment]
I mean training speed. I fully understand what you said, but it currently takes about 30 minutes to train 200 global steps. The number of samples is 400,000 and the 24k model is trained from scratch using one 3090 gpu.

**Comment by OlaWod** [comment]
normally the training speed per batch won't be affected by dataset size. currently i tend to think that maybe there's something wrong in your training code? for example synthesize input features (e.g. spectrogram, ssl feature) on the fly instead of loading from disk.

**Comment by yijingshihenxiule** [comment]
Hello, @lsw5835 ,I met the same question. When I trained with my dataset (vctk+ my own), it cost about 30 minutes train 200 global steps, even 40 minutes. Have you got the solution?
